lib/pbench/server/api/resources/metadata_api.py

```diff
@@ -0,0 +1,355 @@
+            self.logger.warning("Metadata key not provided during metadata query")
+            abort(400, message="Missing metadata key in the query url")
+
+        current_user = self.auth.token_auth.current_user()
+        if current_user:
+            current_user_id = current_user.id
+        else:
+            current_user_id = None
+        print(current_user_id)
+        try:
+            # Fetch the metadata sessions
+            # If the key is favorite, we return only favorite sessions,
+            # else we return all the saved and favorite sessions
+            if key.upper() == "FAVORITE":
+                metadata_sessions = Metadata.query(
+                    user_id=current_user_id, key=MetadataKeys[key.upper()].value
+                )
+            else:
+                metadata_sessions = Metadata.query(user_id=current_user_id)
+            data = [session.get_json() for session in metadata_sessions]
+        except Exception:
+            self.logger.exception(
+                "Exception occurred while querying the Metadata model"
+            )
+            abort(500, message="INTERNAL ERROR")
+
+        response_object = {
+            "message": "success",
+            "data": {"sessions": data},
+        }
+        return make_response(jsonify(response_object), 200)
+
+
+class QueryMetadata(Resource):
+    """
+    Abstracted pbench API for querying a single user metadata session
+    """
+
+    def __init__(self, config, logger):
+        self.server_config = config
+        self.logger = logger
+
+    def verify_metadata(self, session, action):
+        current_user_id = Auth.token_auth.current_user().id
+        metadata_user_id = session.user_id
+        if current_user_id != metadata_user_id:
+            self.logger.warning(
+                "Metadata {}: Logged in user_id {} is different than the one provided in the URI {}",
+                action,
+                current_user_id,
+                metadata_user_id,
+            )
+            abort(403, message="Not authorized to perform the specified action")
+
+    @Auth.token_auth.login_required()
+    def get(self, id=None):
+        """
+        Get request for querying a metadata session of a user given a metadata id.
+        This requires a Pbench auth token in the header field
+
+        The url requires a metadata session id such as /user/metadata/<int:id>
+
+        Required headers include
+            Authorization:   Bearer <Pbench authentication token (user received upon login)>
+
+        :return: JSON Payload
+            response_object = {
+                    "message": "success",
+                    "data": {
+                        "id": "metadata_id",
+                        "value": "client text blob"
+                        "created": "Datetime string"
+                        "updated": "Datetime String"
+                        "key": "key string"
+                        }
+                }
+        """
+        if not id:
```

Comment:
here id would be an int object, isn't `if id is None:` and `if not id:` achieve the same?

Reply:
If `id` is zero, then `not id` will evaluate as `True` whereas `id is None` will evaluate as `False`.  If that's a "feature" (e.g., if an `id` of zero is bad/invalid), then the way you have it coded is OK; otherwise, we're going to lose the first row (if it's ID is `0`).

Mostly I'm pointing out that we need to be conscientious about using arbitrary variables in a boolean context, because, if we mean to be testing for `None`, we might be surprised by what else turns out to be "not True".

Reply:
I don't think the auto-sequenced ID can ever be 0, but the Python semantics point is certainly valid and we should keep that in mind.